FlowCrypt/src/androidTest/java/com/flowcrypt/email/SendMsgTest.kt

```diff
@@ -471,46 +476,61 @@ class SendMsgTest {
           pgpSecretKeyRing = pgpSecretKeyRing
         )
 
-        assertEquals(ENCRYPTED_FORWARDED_ATTACHMENT_NAME, attachmentOpenPgpMetadata.fileName)
+        assertEquals(encryptedForwardedAttachmentInfo.name, attachmentOpenPgpMetadata.fileName)
         assertEquals(true, attachmentOpenPgpMetadata.isEncrypted)
       }
     }
 
     val countOfMsgAfterTest = runBlocking { countOfMsgsOnServer(sentFolder.fullName) }
-    assertEquals(countOfMsgBeforeTest + 1, countOfMsgAfterTest)
+    //as we added 2 messages during this session we use countOfMsgBeforeTest + 2
+    assertEquals(countOfMsgBeforeTest + 2, countOfMsgAfterTest)
   }
 
   private suspend fun checkExistingMsgOnServer(
     folderName: String,
     outgoingMessageInfo: OutgoingMessageInfo,
-    action: suspend (message: MimeMessage) -> Unit
+    useLast: Boolean = true,
+    action: suspend (folder: IMAPFolder, message: MimeMessage) -> Unit
   ) =
     withContext(Dispatchers.IO) {
       val connection = IMAPStoreConnection(context, addAccountToDatabaseRule.account)
+      //need to wait for email server internal sync
+      delay(3000)
       connection.store.use { store ->
         connection.executeIMAPAction {
           store.getFolder(folderName).use { folder ->
             val imapFolder = (folder as IMAPFolder).apply { open(Folder.READ_ONLY) }
-            //get the latest message that we added to a folder recently
-            val mimeMessage = imapFolder.getMessage(imapFolder.messageCount) as MimeMessage
+            //get the message that we added to a folder recently
+            val mimeMessage = (if (useLast) {
+              imapFolder.messages.last()
+            } else {
+              imapFolder.messages.first()
+            }) as MimeMessage
```

Comment:
How about using `getOrElse`? Code length seems similar but `getOrElse` clearly expresses the intention of code i think
```kt
val mimeMessage = imapFolder.messages.getOrElse(
  if (useLast) imapFolder.messages.lastIndex else 0
) { throw NoSuchElementException("List is empty.") } as MimeMessage
```